batcher/aligned-batcher/src/lib.rs

```diff
@@ -384,10 +384,13 @@ impl Batcher {
 
         {
             let mut last_uploaded_batch_block = self.last_uploaded_batch_block.lock().await;
-            self.submit_batch(&batch_bytes, &batch_merkle_tree.root, submitter_addresses)
-                .await;
             // update last uploaded batch block
             *last_uploaded_batch_block = block_number;
+            // This is a hotfix until we merge https://github.com/yetanotherco/aligned_layer/pull/365
+            drop(last_uploaded_batch_block);
+            info!("Batch Finalizer: Last uploaded batch block updated to: {}. Lock unlocked", block_number);
+            self.submit_batch(&batch_bytes, &batch_merkle_tree.root, submitter_addresses)
+                .await;
         }
```

Comment:
it does not make sense to call drop manually, self.submit_batch should be outside {}